File: Android_Client/RegisterClient/app/src/main/java/com/github/meaninglesscode/registerclient/models/api/transaction/types/TransactionEntry.kt
package com.github.meaninglesscode.registerclient.models.api.transaction.types

import android.os.Parcel
import android.os.Parcelable
import com.github.meaninglesscode.registerclient.models.api.constants.PostgresConstants
import com.github.meaninglesscode.registerclient.models.api.transaction.entities.TransactionEntryEntity
import java.util.*

class TransactionEntry(id: UUID = UUID.fromString(PostgresConstants.EMPTY_UUID.value),
                       transactionId: UUID = UUID.fromString(PostgresConstants.EMPTY_UUID.value),
                       lookupCode: String = "", price: Double = -1.0, quantity: Int = -1): Parcelable
{
    var id: UUID = id
    var transactionId: UUID = transactionId
    var lookupCode: String = lookupCode
    var price: Double = price
    var quantity: Int = quantity

    fun toTransactionEntryEntity(): TransactionEntryEntity {
        return TransactionEntryEntity(id.toString(), transactionId.toString(), lookupCode,
                price.toString(), quantity.toString())
    }

    constructor(parcel: Parcel) : this() {
        id = UUID.fromString(parcel.readString())
        transactionId = UUID.fromString(parcel.readString())
        lookupCode = parcel.readString()
        price = parcel.readDouble()
        quantity = parcel.readInt()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(id.toString())
        parcel.writeString(transactionId.toString())
        parcel.writeString(lookupCode)
        parcel.writeDouble(price)
        parcel.writeInt(quantity)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<TransactionEntry> {
        override fun createFromParcel(parcel: Parcel): TransactionEntry {
            return TransactionEntry(parcel)
        }

        override fun newArray(size: Int): Array<TransactionEntry?> {
            return arrayOfNulls(size)
        }
    }
}

File: WebAPI/app/api/transactions/models/types/transactionEntry.ts
const uuid = require("uuidv4");
import { MappedEntityBaseType } from "../../../models/types/mappedEntityBaseType";

export class TransactionEntry extends MappedEntityBaseType {
    public id: string;
    public transactionId: string;
    public lookupCode: string;
    public price: number;
    public discount: number;
    public quantity: number;

    constructor(id = uuid.empty(), transactionId = uuid.empty(), lookupCode = "", price = 0,
        discount = 0, quantity = 0) {
        super(id);
        this.id = id;
        this.transactionId = transactionId;
        this.lookupCode = lookupCode;
        this.price = price;
        this.discount = discount;
        this.quantity = quantity;
    }
}